src/main/capi/appender-c.cpp

```diff
@@ -30,7 +30,7 @@ duckdb_state duckdb_appender_create(duckdb_connection connection, const char *sc
 		wrapper->error = ex.what();
 		return DuckDBError;
 	} catch (...) { // LCOV_EXCL_START
-		wrapper->error = "Unknown create appender error";
+		wrapper->error = "Unknown error in duckdb_appender_create";
```

Comment:
As a note - these `catch(...)` branches are more safeguards against programmer error - they should not happen unless someone throws an exception that does not derive from `std::exception` (e.g. `throw "Error"`).

Reply:
Gotcha. I was wondering about this, also because of the other exception types [here](https://en.cppreference.com/w/cpp/language/try_catch).
```cpp
try
{
    f();
}
catch (const [std::overflow_error](http://en.cppreference.com/w/cpp/error/overflow_error)& e)
{} // this executes if f() throws std::overflow_error (same type rule)
catch (const [std::runtime_error](http://en.cppreference.com/w/cpp/error/runtime_error)& e)
{} // this executes if f() throws std::underflow_error (base class rule)
catch (const [std::exception](http://en.cppreference.com/w/cpp/error/exception)& e)
{} // this executes if f() throws std::logic_error (base class rule)
catch (...)
{} // this executes if f() throws std::string or int or any other unrelated type
```